internal/provider/resource_autogen_subdomain_delegation.go

```diff
@@ -56,41 +64,44 @@ func (r *resourceAutogenSubdomainDelegation) Validate(ctx context.Context, confi
 }
 
 func (r *resourceAutogenSubdomainDelegation) PlanCreate(ctx context.Context, proposed map[string]tftypes.Value, config map[string]tftypes.Value) (map[string]tftypes.Value, []*tfprotov5.Diagnostic, error) {
-	return r.plan(ctx, proposed)
+	// There are never any creates, just set the values to proposed
+	return map[string]tftypes.Value{
+		"id":           proposed["id"],
```

Comment:
```suggestion
		"id":           tftypes.NewValue(tftypes.String, tftypes.Unknown),
```

I'm fairly certain you want this to be set to unknown. This tells Terraform that we don't know what this value will be until after apply. This is true because we need to look up the autogen subdomain to get the id.

Reply:
I have gone ahead and made this change, however, it isn't possible to `create` an autogen_subdomain_delegation. Should we cut this off and return an error in this case?